doc/code/qml_fourier.rst

```diff
@@ -4,6 +4,302 @@ qml.fourier
+
+While all the functions available for visualizing multiple sets of Fourier
+coefficients can be used for a single set, the primary tool for this purpose is
+the ``bar`` function. Using the coefficients we obtained in the
+earlier example,
+
+>>> from pennylane.fourier import *
+>>> import matplotlib.pyplot as plt
+
+>>> fig, ax = plt.subplots(2, 1, sharex=True, sharey=True) # Set up the axes
+>>> bar(coeffs, 1, ax)
+
+.. image:: ../_static/fourier_vis_bar_plot.png
+    :align: center
+    :width: 500px
+    :target: javascript:void(0);
+
+|
+
+In the bar plots, real coefficients are shown in the top panel, and complex in
+the bottom. The labels along the x-axis represent the coefficient frequencies
+(for large plots, it is sometimes convenient to remove these by passing
+``show_freqs=False`` to the plotting function).
+
+Below is a more complex example that demonstrates some of the additional
+customization options available:
+
+.. code::
+
+   from functools import partial
+
+   weights = np.array([[0.1, 0.2, 0.3], [0.4, 0.5, 0.6]])
+
+   @qml.qnode(dev)
+   def circuit_with_weights(w, x):
+       qml.RX(x[0], wires=0)
+       qml.RY(x[1], wires=1)
+       qml.CNOT(wires=[1, 0])
+
+       qml.Rot(*w[0], wires=0)
+       qml.Rot(*w[1], wires=1)
+       qml.CNOT(wires=[1, 0])
+
+       qml.RX(x[0], wires=0)
+       qml.RY(x[1], wires=1)
+       qml.CNOT(wires=[1, 0])
+
+       return qml.expval(qml.PauliZ(0))
+
+   coeffs = coefficients(partial(circuit_with_weights, weights), 2, 2)
+
+   # Number of inputs is now 2; pass custom colours as well
+   fig, ax = plt.subplots(2, 1, sharex=True, sharey=True, figsize=(15, 4))
+   bar(coeffs, 2, ax, colour_dict={"real" : "red", "imag" : "blue"})
+
+.. image:: ../_static/fourier_vis_bar_plot_2.png
+    :align: center
+    :Width: 100%
+    :target: javascript:void(0);
+
+
+Visualizing multiple sets of coefficients
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Suppose we do not want to visualize the Fourier coefficients for a fixed
+``weights`` argument in ``circuit_with_weights``, but the distribution over sets
+of Fourier coefficients when the weights are randomly sampled. For each
+``weights`` sample we get a different set of coefficients:
+
+.. code::
+
+   coeffs = []
+
+   for _ in range(100):
+       weights = np.random.normal(0, 1, size=(2, 3))
+       c = coefficients(partial(circuit_with_weights, weights), 2, degree=2)
+       coeffs.append(np.round(c, decimals=8))
+
+One option to plot the distribution is :func:`~.pennylane.fourier.violin`:
+
+.. code::
+
+   fig, ax = plt.subplots(2, 1, sharey=True, figsize=(15, 4))
+   violin(coeffs, 2, ax, show_freqs=True)
+
+.. image:: ../_static/fourier_vis_violin.png
+    :width: 100%
+    :target: javascript:void(0);
+
+|
+
+A similar option is :func:`~.pennylane.fourier.box`, which
+produces a plot of the same format but using a box plot.
+
+A different view can obtained using the
+:func:`~.pennylane.fourier.radial_box` function. This "rolls up"
```

Comment:
these links aren't resolving 🤔

Reply:
I realize now it is because they are accessible via `qml.fourier.visualize`:
![image](https://user-images.githubusercontent.com/2959003/119678392-0dae2d80-be72-11eb-8b29-930b2ebba715.png)

Reply:
Thanks for catching that! Fixed :+1: